src/components/TransactionFlow.tsx

```diff
@@ -0,0 +1,127 @@
+import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown';
+import { Typography } from '@mui/material';
+import React, { useEffect, useState } from 'react';
+import './transfer-flow.css';
+import { Account } from '../types/chain';
+import { useContractContext } from '../providers';
+import * as polkaUtilsCrypto from '@polkadot/util-crypto';
+import * as polkaUtils from '@polkadot/util';
+
+type ContractType =
```

Comment:
instead of hard coding, can you use the name from the contract abi?

Reply:
The [contractType](https://github.com/hoonsubin/sub0-xvm-workshop/blob/6afa7fe41e010ada3273177780e0f929a34b4814/src/components/TransactionFlow.tsx#L104) state is used for displaying the name of the contract type. I don't think we can find proper text in the [ABI](https://github.com/hoonsubin/sub0-xvm-workshop/blob/6afa7fe41e010ada3273177780e0f929a34b4814/src/abi/evm-erc20.json).  
![image](https://user-images.githubusercontent.com/92044428/203948658-5d4c1eeb-543a-4add-8043-8bcbcd0db845.png)

Reply:
How about changing the name of `ContractType` to `ContractName`?

Reply:
you can read the property `abi.contract.name` and use that as the display name. My point was that the UI shouldn't be aware of the name of the contract, only the abi of the contract that it will be interacting with

Reply:
Please take note:
(1) There is no `abi.contract.name` that can be used from `evm-erc20.json`, so for this, the UI displays hard-coded 'ERC20 Smart Contract'

(2) I've modified displaying `contract name` instead of `contract address`

Reply:
EVM ABI is a bit different. But you can ignore that.
I think this is great!